src/components/PaymentStripe/checkoutForm.js

```diff
@@ -0,0 +1,370 @@
+            ? addressComponents[addressComponents.length - 1]
+            : null;
+        const type = lastAddressComponent ? lastAddressComponent.types[0] : null;
+        if (type === 'postal_code') {
+            const newbillingDetails = {
+                ...billingDetails,
+                ...newDetail,
+                zipCode: lastAddressComponent.long_name,
+            };
+            if (cardElement) {
+                cardElement.update({
+                    value: { postalCode: lastAddressComponent.long_name },
+                });
+            }
+            handleInputChange({ ...newbillingDetails });
+            return;
+        }
+        if (cardElement) {
+            cardElement.update({
+                value: { postalCode: '' },
+            });
+        }
+        handleInputChange({
+            ...newDetail,
+            zipCode: '',
+        });
+    };
+    const handleCardElementChange = event => {
+        setError('');
+        const newbillingDetails = {
+            ...billingDetails,
+            address: {
+                ...billingDetails.address,
+                postal_code: event.value ? event.value.postalCode : '',
+            },
+            zipCode: inline && event.value ? event.value.postalCode : billingDetails.zipCode,
+        };
+        setBillingDetails(newbillingDetails);
+        setCardDetails({ ...event });
+        if (event.error) {
+            setError(event.error.message);
+        }
+        onChange({ billingDetails: newbillingDetails, cardDetails: event });
+    };
+    const handleSubmit = async event => {
+        event.preventDefault();
+        if (!stripe || !elements || !getPaymentIntent || !onProcessed) {
+            return;
+        }
+        try {
+            setLoading(true);
+            // TODO: validate fields and normalize address
+            const response = await getPaymentIntent();
+            const cardElement = inline
+                ? elements.getElement(CardElement)
+                : elements.getElement(CardNumberElement);
+            const result = await stripe.confirmCardPayment(response.client_secret, {
+                payment_method: {
+                    card: cardElement,
+                    billing_details: billingDetails,
+                    setup_future_usage: rememberMe ? 'off_session' : false,
+                },
+            });
+            setLoading(false);
+            if (result.error) {
+                onError(result.error);
+            } else {
+                onProcessed(result);
+            }
+        } catch (e) {
+            setLoading(false);
+            e.type = e.name;
+            onError(e);
+        }
+    };
+
+    return (
+        <Container inline={inline}>
+            <RenderIf isTrue={loading}>
+                <Spinner size="large" />
+            </RenderIf>
+            <form onSubmit={handleSubmit}>
+                <div style={loading ? loadingStyle : null}>
```

Comment:
I'm not sure if we should even show spinner while the stripe SDK is loading or while stripe is resolving the token ... I believe we can just no render anything until stripe is ready ... and disabled the form elements while the stripe is resolving the token